### DISCUSSION TITLE
EKS Fargate with Pod Affinity Rules

### DISCUSSION THREAD

**Discussion by rsmets** [gruntwork-io/knowledge-base#708] [How do I] [answered]
Curious if it is possible to setup EKS with Fargate and have pod affinity rules. The main use of pod affinity rules would be to ensure that the proper distribution of AZs is being used. As far as my understanding goes this sort of configuration is not possible when using EKS Fargate. Anyone think or know to the contrary? 

---

<ins datetime="2023-05-09T22:18:10Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110154">Tracked in ticket #110154</a></p>
</ins>

**Comment by pras111gw** [comment, marked as answer]
Hi Ray, 
You are correct. EKS documentation says affinity/anti-affinity rules do not apply to fargate setup. 
https://www.eksworkshop.com/docs/fundamentals/fargate/enabling/

That said, fargate will automatically distribute pods in a service across all AZs available in a cluster.

**Reply by rsmets** [reply]
Thanks @pras111gw for confirming. I suppose I did not fully realize the notion of the Fargat profile will handle the AZ distribution for you though. Thanks again.
